bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/RemoteControllerWebSocket.java

```diff
@@ -0,0 +1,383 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.eclipse.jetty.util.component.LifeCycle.Listener;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.io.net.http.WebSocketFactory;
+import org.openhab.binding.samsungtv.internal.config.SamsungTvConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link RemoteControllerWebSocket} is responsible for sending key codes to the
+ * Samsung TV via the websocket protocol (for newer TV's).
+ *
+ * @author Arjan Mels - Initial contribution
+ * @author Arjan Mels - Moved websocket inner classes to standalone classes
+ */
+@NonNullByDefault
+public class RemoteControllerWebSocket extends RemoteController implements Listener {
+
+    private final Logger logger = LoggerFactory.getLogger(RemoteControllerWebSocket.class);
+
+    private static final String WS_ENDPOINT_REMOTE_CONTROL = "/api/v2/channels/samsung.remote.control";
+    private static final String WS_ENDPOINT_ART = "/api/v2/channels/com.samsung.art-app";
+    private static final String WS_ENDPOINT_V2 = "/api/v2";
+
+    // WebSocket helper classes
+    private final WebSocketRemote webSocketRemote;
+    private final WebSocketArt webSocketArt;
+    private final WebSocketV2 webSocketV2;
+
+    // JSON parser class. Also used by WebSocket handlers.
+    final Gson gson = new Gson();
+
+    // Callback class. Also used by WebSocket handlers.
+    final RemoteControllerWebsocketCallback callback;
+
+    // Websocket client class shared by WebSocket handlers.
+    final WebSocketClient client;
+
+    // temporary storage for source app. Will be used as value for the sourceApp channel when information is complete.
+    // Also used by Websocket handlers.
+    @Nullable
+    String currentSourceApp = null;
+
+    // last app in the apps list: used to detect when status information is complete in WebSocketV2.
+    @Nullable
+    String lastApp = null;
+
+    // timeout for status information search
+    private static final long UPDATE_CURRENT_APP_TIMEOUT = 5000;
+    private long previousUpdateCurrentApp = 0;
+
+    // UUID used for data exchange via websockets
+    final UUID uuid = UUID.randomUUID();
+
+    // Description of Apps
+    @NonNullByDefault()
+    class App {
+        String appId;
+        String name;
+        int type;
+
+        App(String appId, String name, int type) {
+            this.appId = appId;
+            this.name = name;
+            this.type = type;
+        }
+
+        @Override
+        public String toString() {
+            return this.name;
+        }
+    }
+
+    // Map of all available apps
+    Map<String, App> apps = new LinkedHashMap<>();
+
+    /**
+     * Create and initialize remote controller instance.
+     *
+     * @param host                    Host name of the Samsung TV.
```

Comment:
wrong formatting, see below

Reply:
Sure, I reformatted all files with the ESH codestyle formatter properly enabled.